targets/csharp/source/source/Extensions/Models/PlayFabCloudScriptModelsExtensions.cs.ejs

```diff
@@ -0,0 +1,61 @@
+﻿namespace PlayFab.CloudScriptModels
+{
+    public class FunctionExecutionContext<T>
+    {
+        public PlayFabApiSettings ApiSettings { get; private set; }
+
+        public ProfilesModels.EntityProfileBody CallerEntityProfile { get; set; }
+
+        public T FunctionArgument { get; set; }
+
+        protected FunctionExecutionContext() { }
+
+        public static async System.Threading.Tasks.Task<FunctionExecutionContext<T>> Create(System.Net.Http.HttpRequestMessage request)
+        {
+            string body = await request.Content.ReadAsStringAsync();
+            var contextInternal = Json.PlayFabSimpleJson.DeserializeObject<FunctionExecutionContextInternal>(body);
+            var settings = new PlayFabApiSettings
+            {
+                TitleId = contextInternal.TitleAuthenticationContext.Id,
+                DeveloperSecretKey = contextInternal.TitleAuthenticationContext.SecretKey
+            };
+
+            // TODO: STATIC ENTITY TOKEN?
+            PlayFabSettings.staticPlayer.EntityToken = contextInternal.TitleAuthenticationContext.EntityToken;
+            return new FunctionExecutionContext<T>()
+            {
+                ApiSettings = settings,
+                CallerEntityProfile = contextInternal.CallerEntityProfile,
+                FunctionArgument = (T)contextInternal.FunctionArgument
```

Comment:
This cast doesn't always work. And I suspect that even when it does work, you are not getting what you expect.

Reply:
Please write some tests to verify various different 'Ts' and the various different calling patterns. `FEC<T>.Create, FEC<T>.Create<T>, FEC.Create, FEC.Create<T>`.

Reply:
Fixed as per our conversation: add T on deserialization, remove T from method